cli/src/commands/config/set.rs

```diff
@@ -26,16 +27,21 @@ use crate::cli_util::WorkspaceCommandHelper;
 use crate::command_error::user_error_with_message;
 use crate::command_error::CommandError;
 use crate::complete;
-use crate::config::parse_toml_value_or_bare_string;
+use crate::config::parse_value_or_bare_string;
 use crate::ui::Ui;
 
 /// Update config file to set the given option to a given value.
 #[derive(clap::Args, Clone, Debug)]
 pub struct ConfigSetArgs {
     #[arg(required = true, add = ArgValueCandidates::new(complete::leaf_config_keys))]
     name: ConfigNamePathBuf,
-    #[arg(required = true)]
-    value: String,
+    /// New value to set
+    ///
+    /// The value should be specified in either TOML syntax or string. If a
+    /// string value contained TOML constructs (such as array notation), it must
```

Comment:
Nit: Should be "The value should be specified in either TOML syntax or as a string. If string value contains TOML constructs (such as array notation), it must be quoted".

Reply:
Thanks. Updated the doc.